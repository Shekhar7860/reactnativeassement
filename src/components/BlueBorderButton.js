import React from 'react';
import {View, Text, StyleSheet, TextInput, TouchableOpacity} from 'react-native'
import colors from '../theme/colors';
import commonStyles from '../theme/commonStyles';
const BlueBorderButton = ({title, onPress}) => {
    return (<TouchableOpacity style={styles.blueBorderButton} onPress={onPress}>
           <Text style={{...commonStyles.textStyle, ...styles.textColor}}>{title}</Text>
           </TouchableOpacity>)
}
const styles = StyleSheet.create({
    blueBorderButton : {
    height : 50,
    marginTop : 20,
    backgroundColor : colors.blue,
    justifyContent : 'center',
    width : '80%',
    borderRadius : 10,
    alignSelf : 'center'
    },
    textColor : {
        color : colors.white
    }
})

export default BlueBorderButton;