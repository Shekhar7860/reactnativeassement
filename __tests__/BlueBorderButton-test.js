import React from 'react';
import renderer from 'react-test-renderer';
import BlueBorderButton from "../src/components/BlueBorderButton"

test('renders correctly', () => {
  const tree = renderer.create(<BlueBorderButton />).toJSON();
  expect(tree).toMatchSnapshot();
});