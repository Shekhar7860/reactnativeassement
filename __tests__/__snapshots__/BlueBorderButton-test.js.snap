// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignSelf": "center",
      "backgroundColor": "#6EB1F7",
      "borderRadius": 10,
      "height": 50,
      "justifyContent": "center",
      "marginTop": 20,
      "opacity": 1,
      "width": "80%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#ffffff",
        "fontWeight": "500",
        "textAlign": "center",
      }
    }
  />
</View>
`;
